src/Chunk.cpp

```diff
@@ -1142,11 +1142,18 @@ int cChunk::GrowCactus(int a_RelX, int a_RelY, int a_RelZ, int a_NumBlocks)
 	for (int i = 0; i < ToGrow; i++)
 	{
 		BLOCKTYPE  BlockType;
-		NIBBLETYPE BlockMeta;
-		if (UnboundedRelGetBlock(a_RelX, Top + i, a_RelZ, BlockType, BlockMeta) && (BlockType == E_BLOCK_AIR))
+		if (UnboundedRelGetBlockType(a_RelX, Top + i, a_RelZ, BlockType) && (BlockType == E_BLOCK_AIR))
 		{
-			// TODO: Check the surrounding blocks, if they aren't air, break the cactus block into pickups (and continue breaking blocks above in the next loop iterations)
 			UnboundedRelFastSetBlock(a_RelX, Top + i, a_RelZ, E_BLOCK_CACTUS, 0);
+			if (
+				(UnboundedRelGetBlockType(a_RelX + 1, Top + i, a_RelZ, BlockType) && (BlockType == E_BLOCK_AIR)) ||
+				(UnboundedRelGetBlockType(a_RelX - 1, Top + i, a_RelZ, BlockType) && (BlockType == E_BLOCK_AIR)) ||
+				(UnboundedRelGetBlockType(a_RelX, Top + i, a_RelZ + 1, BlockType) && (BlockType == E_BLOCK_AIR)) ||
+				(UnboundedRelGetBlockType(a_RelX, Top + i, a_RelZ - 1, BlockType) && (BlockType == E_BLOCK_AIR))
```

Comment:
The wiki says "solid material", so things such as torches should be acceptable, too.